### PR TITLE
Backport to v0.10 CI fixes, using GITHUB_TOKEN, change codecov comment layout and rename basecluster to cluster template (#403)

### PR DESCRIPTION
**Pull request by Rohitrajak1807**
* using GITHUB_TOKEN in GH Actions, switch codecov comment layout to diff

* set annotations to false

* change base cluster occurrences to cluster template

* change base cluster occurrence to cluster template in cluster table

* review comments

Aha! Link: https://pf9.aha.io/features/ARLON-363

### CODE REVIEW

File: codecov.yml [hunk 1]
```diff
@@ -6,6 +6,8 @@ coverage:
     patch:
       default:
         informational: true
+github_checks:
+  annotations: false
```
Comment (cruizen) [inline codecov.yml:10]: So this line fixed the annotations (comments that litter the PR reviews)?!

**Review by cruizen — APPROVED** [review]
Don't we need to fix the GitHub token part (and coverage annotations if code coverage is enabled there too) for v0.3 and v0.9 branches too?

**Comment by Rohitrajak1807** [comment]
> Don't we need to fix the GitHub token part (and coverage annotations if code coverage is enabled there too) for v0.3 and v0.9 branches too?

The token bit, yes but not codecov for 0.3

File: codecov.yml (hunk 1 above)
Comment (Rohitrajak1807) [reply codecov.yml:10]: Yes, the docs say so. To actually test it we'll need code changes

**Pull request merged**